go/database/mpt/nodes_test.go

```diff
@@ -6554,6 +6554,162 @@ func markModifiedAsDirty(t *testing.T, ctxt *nodeContext, before, after NodeRefe
 	handle.Release()
 }
 
+func TestVisitPathToAccount_CanIterateNodesByCorrectAddresses(t *testing.T) {
+	address := common.Address{0x12, 0x34, 0x56, 0x78}
+
+	tests := map[string]struct {
+		trie NodeDesc // < the structure of the trie
+		path []string // < the path to follow to reach the test account
+	}{
+		"empty": {
+			trie: &Tag{"A", &Empty{}},
+			path: []string{"A"},
+		},
+		"wrong account": {
+			trie: &Tag{"A", &Account{}},
+			path: []string{"A"}, // < this should be empty
```

Comment:
As the comment says, the path should be empty. The account node in the trie is not the one we are looking for.

The same holds true for the other comments in this test. Is this still work in progress?

Reply:
oh I see now. I did not fully understand the comments, but now I see that the hack to ignore the last element of the path should be removed and instead the path should be shortened here.